### PR TITLE
[MRG+2] Call _transform_vmin_vmax during SymLogNorm.__init__

### PR DESCRIPTION
**Pull request by Salganos**
A fix for #6750

### CODE REVIEW

**Comment by tacaswell** [comment]
:+1: 

Can you also add a test?  It does not need to be an image test, just the minimal code that would have raised before.

**Comment by Salganos** [comment]
Not sure where to add that, but I'll look into it.  The minimal test should be something like:
`
cb.ColorbarBase(plt.figure().add_subplot, norm = clr.SymLogNorm(linthresh = 1, vmin = -1, vmax = 1))
`

File: lib/matplotlib/tests/test_colors.py [hunk 1]
```diff
@@ -222,6 +223,13 @@ def test_SymLogNorm():
     norm = mcolors.SymLogNorm(3, vmin=-30, vmax=5, linscale=1.2)
     normed_vals = norm(vals)
     assert_array_almost_equal(normed_vals, expected)
+    
+    # Ensure that a SymLogNorm object can be used in a colorbar
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_colors.py]: This should be it's own test (any function name that starts with `test_` will work).

It also needs the `@cleanup` decorator.

**Comment by tacaswell** [comment]
:heart: pep8/pycodestyle 

There is some trailing white space:

```
FAIL: matplotlib.tests.test_coding_standards.test_pep8_conformance_installed_files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/tests/test_coding_standards.py", line 244, in test_pep8_conformance_installed_files
    expected_bad_files=expected_bad_files)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/tests/test_coding_standards.py", line 143, in assert_pep8_conformance
    assert_equal(result.total_errors, 0, msg)
AssertionError: 1 != 0 : Found code syntax errors (and warnings):
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/tests/test_colors.py:226:1: W293 blank line contains whitespace

```

This may seem petty and annoying, but keeping a consistent style (including no trailing white space) really does help on big projects with many contributors.

**Comment by Salganos** [comment]
Whoops.  The perils of not using a proper development environment.

File: lib/matplotlib/tests/test_colors.py [hunk 2]
```diff
@@ -224,6 +225,17 @@ def test_SymLogNorm():
...
 
+@cleanup
+def test_SymLogNorm_colorbar():
+    """
+    Test un-called SymLogNorm in a colorbar.
+    """
+    norm = mcolors.SymLogNorm(0.1, vmin=-1, vmax=1, linscale=1)
+    fig = plt.figure()
+    cbar = mcolorbar.ColorbarBase(fig.add_subplot(111), norm=norm)
+    plt.close(fig)
```
Comment (efiring) [inline lib/matplotlib/tests/test_colors.py]: I think this `close` is unnecessary; the `@cleanup` decorator handles it, doesn't it?

Comment (tacaswell) [inline lib/matplotlib/tests/test_colors.py]: It does, but this does not _hurt_

**Comment by tacaswell** [comment]
Changed my mind about targeting 2.x .  If someone disagrees, I will not protest to it being backported.

**Comment by efiring** [comment]
It's not critical, but I am mildly in favor of the backport.  The change has no effect in most uses, but blocks a potential failure mode.

**Comment by QuLogic** [comment]
@efiring Are you going to backport?

**Comment by efiring** [comment]
Sure, I can do that in a few minutes.

**Comment by efiring** [comment]
Backported to v2.x as 0ab4599

**Pull request merged**